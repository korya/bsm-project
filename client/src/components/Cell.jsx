import React from 'react';

export default function Cell({ number, selected, onClick }) {
  const className = selected ? 'selected' : '';

  return (
    <td>
      <button
        className={className}
        onClick={onClick}
      >
        {number}
      </button>
    </td>
  );
}
