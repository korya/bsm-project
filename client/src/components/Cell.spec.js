import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Cell from './Cell';

test('renders correctly', () => {
  expect(
    renderer.create(<Cell number={1} />).toJSON(),
  ).toMatchSnapshot();
  expect(
    renderer.create(<Cell number={9} />).toJSON(),
  ).toMatchSnapshot();
  expect(
    renderer.create(<Cell selected={true} number={7} />).toJSON(),
  ).toMatchSnapshot();
});

test('calls onClick prop handler', () => {
  const clickHandler = jest.fn();
  const wrapper = shallow(<Cell onClick={clickHandler} />);
  wrapper.find('button').simulate('click');
  expect(clickHandler).toHaveBeenCalledTimes(1);
});
