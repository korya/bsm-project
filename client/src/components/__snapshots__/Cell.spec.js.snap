// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<td>
  <button
    className=""
  >
    1
  </button>
</td>
`;

exports[`renders correctly 2`] = `
<td>
  <button
    className=""
  >
    9
  </button>
</td>
`;

exports[`renders correctly 3`] = `
<td>
  <button
    className="selected"
  >
    7
  </button>
</td>
`;
